pkg/gatherers/clusterconfig/gather_cluster_roles.go

```diff
@@ -0,0 +1,66 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+)
+
+// GatherClusterRoles Collects definition of the "admin" and "edit" cluster roles.
+//
+// ### API Reference
+// - https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/rbac/types.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/cluster-scoped-resources/rbac.authorization.k8s.io/clusterroles
+//
+// ### Location in archive
+// - `cluster-scoped-resources/rbac.authorization.k8s.io/clusterroles/`
+//
+// ### Config ID
+// `clusterconfig/clusterroles`
+//
+// ### Released version
+// - 4.18.0
+//
+// ### Backported versions
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherClusterRoles(ctx context.Context) ([]record.Record, []error) {
+	kubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherClusterRoles(ctx, kubeClient.RbacV1(), []string{"admin", "edit"})
+}
+
+func gatherClusterRoles(ctx context.Context, rbacV1Cli v1.RbacV1Interface, names []string) ([]record.Record, []error) {
+	var errs []error
+	var records []record.Record
+	for _, name := range names {
+		clusterRoleRec, err := gatherClusterRole(ctx, name, rbacV1Cli)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
```

Comment:
I usually prefer to have the conditional with the error then returning/continuing the loop. That way the rest of the code lacks the need for tabulation.

e.g.
```
if err != nil {
	errs = append(errs, err)
	continue
}
records = append(records, *clusterRoleRec)
```
But this is not a blocker.